.github/workflows/scala-publish

```diff
@@ -0,0 +1,20 @@
+name: Publish Jar to Sonatype
+on:
+  release:
+    types: [created]
```

Comment:
guess my question is that how these test can fail if we did not bump version number from `0.0.4-SNAPSHOT` to `0.0.4` ?

nit tho.

Reply:
So the idea is that we'll use `bump2version` to increment the python version from say 0.0.4dev0 to 0.0.4 and create a v0.0.4 tag. sbt dynver will then use the tag to determine that this is 0.0.4 jar (non-snapshot since it's tagged). 

For conftest and notebooks, I will add something else to automatically update those references.

The main way this could go wrong is if we create a release on a commit without a tag (e.g., we forgot to add a tag, we pointed to the wrong commit, we forgot to bump version and somehow added the tag manually). That would cause sbt to publish a snapshot build to the staging repo and pypi to get a dev0 build. In that scenario I believe we can just fix master and then create another release. It would re-trigger the two release workflows.

Another possibility is if we didn't wait for the tagged commit to be green and it failed later. In that case we may have to just bump the version again and mark a particular release as broken or something